src/state/window.ts

```diff
@@ -0,0 +1,67 @@
+import { BrowserWindow, Rectangle } from 'electron'
+import appConfig from 'electron-settings'
+
+interface State {
+  bounds?: Rectangle
+  isFullScreen?: boolean
+  isMaximized?: boolean
+}
+
+export default class WindowState {
+  private _state: State
```

Comment:
This is personal preference, but how about omitting the underscore prefix, because the property is already declared as `private`? Same for the rest below.

Reply:
That was one of the rules in xo TypeScript (`@typescript-eslint/member-naming`).  I can disable if you like.  I would agree that it doesn't look that great.

Reply:
I just had a quick look for the reasoning to use `_`: https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/member-naming.md

> For example, prefixing private properties with a `_` allows them to be easily discerned when being inspected by tools that do not have knowledge of TypeScript (such as most debuggers).

I think it's a reason I can live with.